pkg/ui/json.go

```diff
@@ -0,0 +1,71 @@
+package ui
+
+import (
+	"github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/anz-bank/sysl/pkg/syslutil"
+)
+
+const GRPC = "grpc"
+const REST = "rest"
+
+// WebService is the struct which will be translated to JSON via reflection.
+type WebService struct {
+	Name       string
+	Attributes map[string]string
+	Endpoints  []Endpoint
+	Type       string
+	Path       string
+}
+
+// BuildWebService takes a sysl Application and returns a json-exportable representation of Sysl
+func BuildWebService(a *sysl.Application) (*WebService, error) {
+	// TODO: Ensure there are no duplicate service names.
+	serviceMethod := serviceType(a)
+	serviceName := a.GetName().GetPart()[0]
+	attributes := mapAttributes(a.GetAttrs())
+	endpoints := mapEndpoints(a.GetEndpoints())
```

Comment:
Why don't populate the Webservice object directly

Reply:
Good point, I thought it looked cleaner but now that I look at it again, I think I prefer doing it directly. Saves 5 extra lines of code as well!